docs/contribution-guides/Development-Standards-Guide.md

```diff
@@ -226,3 +242,307 @@ to just leave it as a public field without a property. As always, make sure to d
+{
+    public bool StateChanged;
+    public int currentHealth = 5;
+
+    void stateChange(int damageTaken){
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        StateChanged = true;
+        Inventory.DropAll(); //How do we know that we have an inventory at all?
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    public Health health; //<- possibily null
+
+    void Awake(){
+        UpdateManager.Add(FIXED_UPDATE, myFunc);
+    }
+
+    void myFunc(){
+        if(health.StateChanged == false) return;
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory{
+    public void DropAll();
+}
+```
+
+```cs
+//example of good code
+
+class Health : Monobehavior
+{
+    public int CurrentHealth {private set; get;} = 5;
+    public Action OnDeath = new Action();
+
+    public void TakeDamage(int damageTaken)
+    {
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        OnDeath?.Invoke();
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    void OnEnable()
+    {
+        if(gameObject.TryGetComponent<Health>(out var playerHealth) == false) 
+        {
+            // Always leave warnings and errors whenever possible instead of silently not reporting unintended behaviors.
+            Logger.LogWarning("[DeathUI/OnEnable] - No health found on this gameObject! DeathUI hasn't subscribed to any events when it got enabled!");
+            return;
+        }
+        // Subscribes to the OnDeath event which will trigger when this object's health reaches 0 or below.
+        playerHealth.OnDeath += myFunc;
+    }
+
+    void myFunc()
+    {
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory
+{
+    //same thing as in DeathUI, just subscribe to the OnDeath event.
+}
+```
+
+_Note: Events can leak! Always remember to unsubscribe your events on `OnDisable()`._
+
+* Minimise inheritance, Embrace Interfaces.
+
+[Interface](https://www.w3schools.com/cs/cs_interface.php) in C# is a blueprint of a class. It is like abstract class because all the methods which are declared inside the interface are abstract methods. It cannot be instantiated, but it can hold optional default implamentations. It is used to achieve multiple inheritance which can't be achieved by class.
+
+Interfaces allows to share behaviour with classes in a horizontal manner, contrary to inheritance where the sharing is vertical (from parent to children). In horizontal, there is no need for a parental relationship between the classes. 
+
+_CodeMonkey has a great video on how to utilise interfaces for a modular workflow: [Modular Character System in Unity](https://youtu.be/mJRc9kLxFSk)_
+
+That being said, Concepts such as [Polymorphism](https://www.w3schools.com/cs/cs_polymorphism.php) is not something to be afraid of. Inheritance makes sense to be used while in the right context, which is mostly going to be in ScriptableObjects and scanerios where you want to work with common data that will be shared around. 
```

Comment:
you are still linking to polymorphism here. The concept you should not be afraid of is inheritance. Both interfaces and class inheritance achieve polymorphism.

Reply:
Done 👍